node/core/candidate-validation/src/lib.rs

```diff
@@ -306,18 +310,25 @@ where
 			},
 		};
 
-	let validation_code = match sp_maybe_compressed_blob::decompress(
+	let executor_params =
+		if let Ok(executor_params) = executor_params_at_relay_parent(relay_parent, sender).await {
+			executor_params
+		} else {
+			return PreCheckOutcome::Invalid
```

Comment:
I don't remember our previous discussion on this topic anymore. But how's that PVF's fault that we can't fetch executor params for some reason? These params are only set by the relay chain. I can think of two scenarios of how the error here could happen (please correct me if I'm wrong):
1. The relay_parent was pruned. E.g. we finalized a different block at the same height.
2. We're shutting down.

In either case, why would we be voting against the PVF itself?

Reply:
Yeah, I think it's a valid point. If we're shutting down, we shouldn't vote against it. However, I still need to figure out the first case. The relay parent is pruned, we cannot obtain the session index for that relay parent, so the error will persist no matter how much we retry the request. That sounds like a deterministic error, so the PVF should be considered invalid. But I'm not sure I'm not missing something.

Reply:
@mrcnski, you were sorting out those deterministic vs. non-deterministic error issues, what do you think?

Reply:
I would say this should be `Failed`, as it's not an issue with the execution of the PVF itself. cc @eskimor just in case. But luckily, it doesn't matter too much as we now [reject failed PVFs](https://github.com/paritytech/polkadot/pull/6492) as well as invalid ones, so we should not retry the request. We should still add a `warn` log here, IMO.

Reply:
If in doubt vote against. If it is a local/transient issue on one validator, it does not matter if it votes invalid - there will still be a super majority approving the PVF. If not, then we have a bug that needs to be fixed: Abstaining from voting would not help here either - as all/other preparation jobs will fail as well. So we got a problem regardless.